File: BikeAI/Simulation/SugPair.py
from BikeAlg import *
from Station import *
from EndLocation import *
from DisPair import *
from SugPair import *

class SugPair:
    def __init__(self, station, sug):
        self.station = station
        self.sug = sug

    def compare(self, pair):
        if (self.sug - pair.sug > 0):
            return 1
        if (self.sug - pair.sug == 0):
            return 0
        return -1

    def __lt__(self, other):
        return self.sug < other.sug

    def __gt__(self, other):
        return self.sug > other.sug

    def __eq__(self, other):
        return self.sug == other.sug

    def __ne__(self, other):
        return not self.__eq__(other)

    def __str__(self):
        return "(" + str(self.station.id) + ", " + str(self.sug) + ")"

File: Simulation/src/SugPair.java

public class SugPair implements Comparable<SugPair>{
	 public Station station;
	  public double sug;
	  
	  public SugPair(Station station, double sug){
		    this.station = station;
		    this.sug = sug; 
		  }
	  
	  public int compareTo(SugPair pair){
		  
		  if(this.sug - pair.sug > 0) {
			  return 1;
		  }
		  if(this.sug- pair.sug == 0) {
			  return 0;
		  }
		  return -1;
	  }
	  
	  public String toString() {
		  String tostring =  "(" + station.id + ", " + sug + ")";
		  return tostring;
	  }
}
